pkg/devfile/parser/context/context.go

```diff
@@ -52,22 +75,25 @@ func (d *DevfileCtx) Populate() (err error) {
 		return err
 	}
 
-	// Get devfile APIVersion
-	if err := d.SetDevfileAPIVersion(); err != nil {
+	if err := d.PopulateDevfile(); err != nil {
```

Comment:
This and the following lines could be simplifies to simply: `return d.PopulateDevfile()`

Reply:
Good shout! thanks :)